proxy/pkg/proxy/proxy.go

```diff
@@ -124,6 +125,7 @@ func (p *MultiHostReverseProxy) directorSelectionDirector(r *http.Request) {
 		return
 	}
 
+	method := ""
```

Comment:
I can't fully wrap my head around this at the moment but shouldn't this be defined inside of the loop or at least be reset to "" at the beginning of the loop?